app/gui/src/project-view/providers/tooltipRegistry.ts

```diff
@@ -46,20 +52,32 @@ export const [provideTooltipRegistry, useTooltipRegistry] = createContextStore(
         const methods = {
           onTargetEnter(target: HTMLElement) {
             const entriesSet: EntriesSet = hoveredElements.get(target) ?? shallowReactive(new Set())
-            entriesSet.add(entry)
+            entriesSet.add({ contents: slot, isHidden: false, key })
             // make sure that the newly entered target is on top of the map
             hoveredElements.delete(target)
             hoveredElements.set(target, entriesSet)
             registeredElements.add(target)
           },
           onTargetLeave(target: HTMLElement) {
             const entriesSet = hoveredElements.get(target)
-            entriesSet?.delete(entry)
+            if (entriesSet) {
+              for (const e of entriesSet) {
+                if (e.key === key) entriesSet.delete(e)
+              }
+            }
```

Comment:
We may have quite a lot of entries in this set - could it be a map?

Also, why didn't the previous approach work? In general I feel a bit more documentation would be helpful; for example I don't get why `hoveredElements` is map of HTMLElement  -> Set - do we have many tooltips registered for single HTML element?